### ISSUE TITLE
Harpoon requires telescope when enableTelescope is set to false.

### ISSUE THREAD

**Issue by adalessa** [nix-community/nixvim#658]
**Plugin affected**: harpoon, telescope

## Description
Having installed telescope makes require to enable telescope in harpoon `plugins.harpoon.enableTelescope`

## Config
```nix
plugins.telescope = {
    enable = true;
}

plugins.harpoon = {
    enable = true;
    enableTelescope = false;
}
```
This fails with the error 
```
       error:
       Failed assertions:
       - Nixvim: The harpoon telescope integration needs telescope to function as intended
```
I think the problem is here 
https://github.com/nix-community/nixvim/blob/main/plugins/utils/harpoon.nix#L205
But the expectation should be that if you enableTelescope requires that enables telescope, not the other way arround.

**Comment by GaetanLepage** [comment]
Can you try to update your flake and see if the problem occurs ?
We have fixed it in #656

**Comment by adalessa** [comment]
Thanks, now is working as expected. Thanks.
